wfl-slack-app/src/wfl-api.js

```diff
@@ -0,0 +1,98 @@
+var request = require('request');
+
+function postFoodOrder(restaurantId, callback) {
+    let options = {
+        url: process.env.DB_URI + '/food-order',
+        form: {
+            RestaurantId: restaurantId
+        }
+    }
+
+    request.post(options, function(err, res, body) {
+        let json = JSON.parse(body);
+        if (json) {
+            callback(json);
+        }
+    });
+}
+
+function getAllRestaurants(callback) {
+    let options = {
+        url: process.env.DB_URI + '/restaurant'
+    }
+    request.get(options, function(err, res, body) {
+        callback(JSON.parse(body));
+    });
+}
+
+function getFoodOrderById(id, callback) {
+    let options = {
+        url: process.env.DB_URI + '/food-order/' + id
+    }
+    request.get(options, function(err, res, body){
+        callback(JSON.parse(body));
+    });
+}
+
+function getRestaurantByName(name, callback) {
+    let options = {
+        url: process.env.DB_URI + '/restaurant' + '?Name=' + name,
+    }
+
+    request.get(options, function(err, res, body) {
+        let json = JSON.parse(body);
+        if(json.entities && json.entities[0]) {
+            callback(json.entities[0]);
+        }
+        else {
+            callback();
+        }
+    });
+}
+
+module.exports = {
+    getRestaurantByName: getRestaurantByName,
+    postFoodOrder: postFoodOrder,
+    getAllRestaurants: getAllRestaurants,
+    getFoodOrderById: getFoodOrderById,
+    postMenuSelection: function(foodOrderId, name, description, callback) {
+        let options = {
+            url: process.env.DB_URI + '/food-order/' + foodOrderId + '/menu-selection',
```

Comment:
I think this is where we'll have to make a decision.  Hypermedia and proper Siren understanding should help a bit here, as it completely removes the need to combine URLs like this.  But at the same time, I want to have something that fully works before you are finished.  I'm thinking it'll make sense to get this working as is, get it merged, then dig into leveraging hypermedia from a client perspective.

Reply:
I agree with this. It would be nice to see hypermedia leveraged both at the web api we created and in the slack app that sends request to it as well. I also agree that it may be best to get this PR merged first so that we have something that works going before my internship ends.